projects/js-packages/publicize-components/src/components/media-section-v2/index.tsx

```diff
@@ -183,6 +187,37 @@ export default function MediaSectionV2( {
 		}, 0 );
 	}, [] );
 
+	// Handle AI image generation click
+	const handleAiImageClick = useCallback( () => {
+		setShowAiImageModal( true );
+	}, [] );
+
+	// Handle AI image modal close
+	const handleAiImageModalClose = useCallback( () => {
+		setShowAiImageModal( false );
+	}, [] );
```

Comment:
You won't need to define these two separately if you use `useReducer` above instead of `useState`.

```ts
const [ showAiImageModal, toggleShowAiImageModal ] = useReducer( ( state ) => ! state, false );
```

You can then directly pass `toggleShowAiImageModal` for both open and close callbacks.